src/main/java/stsjorbsmod/patches/BottledMemoryPatch.java

```diff
@@ -0,0 +1,113 @@
+package stsjorbsmod.patches;
+
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpireField;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePostfixPatch;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.helpers.RelicLibrary;
+import com.megacrit.cardcrawl.relics.AbstractRelic;
+import com.megacrit.cardcrawl.relics.BottledFlame;
+import com.megacrit.cardcrawl.relics.BottledLightning;
+import com.megacrit.cardcrawl.relics.BottledTornado;
+import javassist.CannotCompileException;
+import javassist.expr.ExprEditor;
+import javassist.expr.FieldAccess;
+import javassist.expr.MethodCall;
+import stsjorbsmod.relics.BottledMemoryRelic;
+
+public class BottledMemoryPatch {
+    @SpirePatch(clz = AbstractCard.class, method = SpirePatch.CLASS)
+    public static class AbstractCardMemoryFields {
+        public static SpireField<Boolean> inBottleMemory = new SpireField<>(() -> false);
+    }
+
+    @SpirePatch(clz = AbstractCard.class, method = "makeStatEquivalentCopy")
+    public static class AbstractCard_makeStatEquivalentCopy {
+        @SpirePostfixPatch
+        public static AbstractCard patch(AbstractCard __return, AbstractCard __this) {
+            AbstractCardMemoryFields.inBottleMemory.set(__return, AbstractCardMemoryFields.inBottleMemory.get(__this));
+            return __return;
+        }
+    }
+
+    public static class BottledGetCardsExprEditor extends ExprEditor {
+        private String methodName;
+
+        BottledGetCardsExprEditor(String methodName) {
+            this.methodName = methodName;
+        }
+
+        @Override
+        public void edit(MethodCall methodCall) throws CannotCompileException {
+            if (methodCall.getClassName().equals(CardGroup.class.getName()) && methodCall.getMethodName().equals(methodName)) {
+                methodCall.replace("{ $_ = (stsjorbsmod.patches.BottledMemoryPatch.removeBottledCards($proceed())); }");
+            }
+        }
+    }
+
+    /**
```

Comment:
This comment doesn't add any information that the method signature doesn't already cover